x-pack/platform/plugins/shared/observability_ai_assistant/server/service/distributed_lock_manager/index.ts

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Client, errors } from '@elastic/elasticsearch';
+import { Logger } from '@kbn/logging';
+import pRetry from 'p-retry';
+
+const INDEX_NAME = '.kibana-distributed-lock-manager';
+
+export enum LockId {
+  KnowledgeBaseReindex = 'knowledge_base_reindex',
+}
+
+export interface LockDocument {
+  createdAt: string;
+  expiresAt: string;
+  meta: Record<string, any>;
+}
+
+interface AcquireOptions {
+  meta?: Record<string, any>;
+  ttlMs?: number;
+}
+
+export class LockManager {
+  constructor(private lockId: LockId, private esClient: Client, private logger: Logger) {}
```

Comment:
I don't know if there will be some use case for using different lockId within a single request, and therefore the need for instantiating many LockManager. 
But we could potentially move the lockId within the acquire function. Or have two, one used as a prefix, one as a suffix to avoid conflict from various consumers?

Reply:
Handling multiple locks is a valid use case and imo should be done with separate `lockManager` instances. An instance comes with a unique token and I imagine a lot of trouble if we re-use the same token for different locks.

Reply:
I've made `LockManager` private, so now only the `withLock` method is exposed. Let's focus on the ergonomics of using that for now.